### PR TITLE
drivers/pinmux: stm32f1: Rework remap field encoding

### PR DESCRIPTION
**Pull request by erwango**
Suggested by @gmarull in https://github.com/zephyrproject-rtos/zephyr/pull/40194#issuecomment-972648478.

Aim of the change is to stop requiring device reg address information in STM32F1 pinmux api.
This is done by adding all the information required for pin remap configuration directly inside remap
field which is provided in pinctrl device tree description.

This is an alternate solution to https://github.com/zephyrproject-rtos/zephyr/pull/40430,
which was raised to enable new pinctrl API deployment in case of STM32F1 PWM (issue described [here](https://github.com/zephyrproject-rtos/zephyr/pull/40194#issuecomment-965031732).

### CODE REVIEW

**Review by mbolivar-nordic — APPROVED** [review]

File: include/dt-bindings/pinctrl/stm32f1-afio.h [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+#define USART2_REMAP0		STM32_REMAP(0U, 0x1U, 3U, STM32_AFIO_MAPR)
+/** USART2 (remap) */
+#define USART2_REMAP1		STM32_REMAP(1U, 0x1U, 3U, STM32_AFIO_MAPR)
+
+/** USART3 (no remap) */
+#define USART3_REMAP0		STM32_REMAP(0U, 0x3U, 4U, STM32_AFIO_MAPR)
+/** USART3 (partial remap) */
+#define USART3_REMAP1		STM32_REMAP(1U, 0x3U, 4U, STM32_AFIO_MAPR)
+/** USART3 (full remap) */
+#define USART3_REMAP1		STM32_REMAP(1U, 0x3U, 4U, STM32_AFIO_MAPR)
```
Comment (gmarull) [inline include/dt-bindings/pinctrl/stm32f1-afio.h:108]: Duplicate definition here?

File: include/dt-bindings/pinctrl/stm32f1-afio.h [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+ *
+ * - 0:    reg (0 or 1).
+ * - 1..5: shift (0..31).
+ * - 6..7: mask (0x1, 0x3).
+ * - 8..9: val (0..3).
+ *
+ * @param reg AFIO_PCFx register (0, 1).
+ * @param shift Position within AFIO_PCx.
+ * @param mask Mask for the AFIO_PCx field.
+ * @param val Remap value (0, 1, 2 or 3).
```
Comment (gmarull) [inline include/dt-bindings/pinctrl/stm32f1-afio.h:21]: AFIO_PCx -> AFIO_MAPR/2

File: include/dt-bindings/pinctrl/stm32f1-afio.h (hunk 1 above)
Comment (erwango) [reply include/dt-bindings/pinctrl/stm32f1-afio.h:108]: Yes, fixed, Txs

File: include/dt-bindings/pinctrl/stm32f1-afio.h (hunk 2 above)
Comment (erwango) [reply include/dt-bindings/pinctrl/stm32f1-afio.h:21]: Oops, fixed

**Review by gmarull — APPROVED** [review]

**Comment by erwango** [comment]
@carlescufi this one is ready for merge.

**Review by FRASTM — APPROVED** [review]

**Pull request merged**